foodalert/test/test_allergen.py

```diff
@@ -0,0 +1,172 @@
+from django.db import connection, transaction
+
+import foodalert
+from foodalert.models import Allergen
+from foodalert.serializers import AllergenSerializer
+from foodalert.views import AllergensList
+
+class AllergenTest(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Sets up a single mock user that can be used for all tests on
+        notification tests
+        """
+        user = "testuser"
+        passw = "test"
+        cls.user = User.objects.create_user(username=user,
+                                            email="testuser@test.com",
+                                            password=passw,
+                                            is_active=1)
+        cls.realAllergen = Allergen.objects.create(
+                name= "test allergen"
+                )
+
+    def setUp(self):
+        self.client.force_login(self.user)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete()
+        cls.realAllergen.delete()
+
+    def tearDownn(self):
+        pass
+
+    def test_get_allergen_list(self):
+        """
+        This tests that you can read all allergens
+        in the db from a get request. Should return a 
+        200 response upon success
+        """
+        response = self.client.get('/allergen/')
+        self.assertEqual(200, response.status_code)
+        data = response.data[0]
+        self.assertEqual(data["name"], self.realAllergen.name)
+    
+    def test_get_allergen_detail(self):
+        """
+        This tests that you can read allergen detail
+        corresponding to the id
+        """
+        id = self.realAllergen.id
+        response = self.client.get('/allergen/{}/'.format(id))
+        self.assertEqual(200, response.status_code)
+        data = response.data
+        self.assertEqual(self.realAllergen.name, data["name"])
+
+    def test_post_allergen(self):
+        """
+        This tests that an allergen should be created 
+        correctly from post request. Should return a 201
+        response upon success
+        """
+        valid_payload = {
+            "name": "wheat"
+        }
+        original_len = len(Allergen.objects.all())
+
+        response = self.client.post('/allergen/', valid_payload)
+        self.assertEqual(201, response.status_code)
+        new_len = len(Allergen.objects.all())
+        self.assertEqual(1, new_len - original_len)
+
+    def test_post_existing_allergen(self):
+        """
+        Test should return a 400 bad request error if post request contains an 
+        allergen name that already exists in the database. New object should not
+        be made
+        """
+        original_len = len(Allergen.objects.all())
+        payload = {
+            "name": self.realAllergen.name
+        }
+        response = self.client.post('/allergen/', payload)
+        self.assertEqual(400, response.status_code)
+        after_len = len(Allergen.objects.all())
+        self.assertEqual(original_len, after_len)
```

Comment:
Probably additionally want to do a GET and check the value(s) of your returned object.